component/src/main/java/org/wso2/extension/siddhi/io/cdc/util/CDCSourceUtil.java

```diff
@@ -0,0 +1,237 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.cdc.util;
+
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.wso2.extension.siddhi.io.cdc.source.InMemoryOffsetBackingStore;
+import org.wso2.extension.siddhi.io.cdc.source.WrongConfigurationException;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class contains Util methods for the cdc extension.
+ */
+public class CDCSourceUtil {
+    public static Map<String, Object> getConfigMap(String username, String password, String url, String tableName,
+                                                   String historyFileDirectory, String siddhiAppName,
+                                                   String siddhiStreamName, int serverID, String serverName,
+                                                   String connectorProperties, int cdcSourceHashCode)
+            throws WrongConfigurationException {
+
+        Map<String, Object> configMap = new HashMap<>();
+
+        String host;
+        int port;
+        String database;
+
+        //Add schema specific details to configMap
+
+        String[] splittedURL = url.split(":");
+        if (!splittedURL[0].equalsIgnoreCase("jdbc")) {
+            throw new WrongConfigurationException("Invalid JDBC url: " + url);
+        } else {
+            switch (splittedURL[1]) {
+                case "mysql": {
+
+                    //Extract url details
+                    String regex = "jdbc:mysql://(\\w*|[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}):" +
+                            "(\\d++)/(\\w*)";
+                    Pattern p = Pattern.compile(regex);
+                    Matcher matcher = p.matcher(url);
+                    if (matcher.find()) {
+                        host = matcher.group(1);
+                        port = Integer.parseInt(matcher.group(2));
+                        database = matcher.group(3);
+
+                    } else {
+                        throw new WrongConfigurationException("Invalid JDBC url: " + url);
```

Comment:
Invalid JDBC url: " + url for source ''' expected in the format of

Reply:
```java
throw new WrongConfigurationException("Invalid JDBC url: " + url + " for cdc source");
```
Is this correct? or should also I specify the respective stream name?

Reply:
Lets log the stream name too.. Else no body can figure out which source it throwing error. Also give the expected format

Reply:
Fixed with commit: `Add stream details to invalid JDBC url error.` f2ad73e